### DISCUSSION TITLE
E2setupRequest malformed packet

### DISCUSSION THREAD

**Discussion by oliferch** [srsran/srsRAN_Project#476] [General Help]
Hello srsRAN Team,

i faced an issue with the E2setupRequest send by the E2 agent. The message contains different IEs such as Global E2 Node Id, Transaction Id, Ran Functions Added list, E2 Node Component Configuration Addition List (screenshot attached for your reference).

![E2setupRequest](https://github.com/srsran/srsRAN_Project/assets/63338515/c422e4ea-0e93-481c-8d3d-b9a57e82127f)

As per the attached pcap it seems E2 Node Component Configuration Addition List IE is not filled properly by the srsRAN e2 agent and because of this a E2setupFailure is send by the RIC.

[e2ap.zip](https://github.com/srsran/srsRAN_Project/files/14297278/e2ap.zip)

Many thanks for your great work. Appreciate any feedback on this topic.

Regards,
Oliver

**Comment by pgawlowicz** [comment]
Hi @oliferch,

Indeed, the second part of the E2 Setup Request (i.e. Component Configuration Addition List) is not filled properly yet. We are working on that. Did you get any issues with RIC rejecting the message due to this issue? If so, could you tell which RIC do you use?

Piotr

**Reply by oliferch** [reply]
Hi Piotr,

thanks for your super fast feedback and the information that you work on that topic already.
It is not really an issue as the RIC replies correctly to the E2 Setup request. I am using a COTS RIC test solution from Keysight

It would be nice if you can write a short reply once the issue is fixed.

Thank you !

Regards,
Oliver

**Reply by pgawlowicz** [reply]
Hi @oliferch, 
the e2 setup message is now asn1 compatible. However, currently it is filled with a dummy data, we will work on this in the following weeks.

**Reply by oliferch** [reply]
Hi Piotr,

that is great. I will check it soon.
Many thanks for your great work once again.

Regards,
Oliver
